Sources/Misc/SystemInfo.swift

```diff
@@ -33,6 +33,7 @@ class SystemInfo {
     }
 
     let storeKit2Setting: StoreKit2Setting
+    let storeKitVersion: StoreKitVersion
```

Comment:
Per my other comment, this is the main thing I'm concerned about: we now have 2 sources of truth.

I'd remove `StoreKit2Setting`, and instead derive the existing logic from the new `StoreKitVersion`.